sound/soc/sof/intel/hda-loader.c

```diff
@@ -407,11 +407,15 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 	 * should be ready for code loading and firmware boot
 	 */
 	ret = cl_copy_fw(sdev, stream);
-	if (!ret)
+	if (!ret) {
 		dev_dbg(sdev->dev, "Firmware download successful, booting...\n");
-	else
+	} else {
 		dev_err(sdev->dev, "error: load fw failed ret: %d\n", ret);
 
+		/* dump dsp registers and disable DSP upon error */
```

Comment:
The "and disable DSP upon error" comment is a bit misleading when moved here maybe remove the comment altogether, the code is pretty self-explanatory